### PR TITLE
QUANTUM ESPRESSO

### PR DESCRIPTION
**Pull request by jan-janssen**
@conda-forge/help-c-cpp

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
So it seems to be an issue with the linker:
```
/home/conda/feedstock_root/build_artifacts/qe_1562835890875/_build_env/bin/x86_64-conda_cos6-linux-gnu-ld: unrecognized option '-Wl,-O2'
/home/conda/feedstock_root/build_artifacts/qe_1562835890875/_build_env/bin/x86_64-conda_cos6-linux-gnu-ld: use the --help option for usage information
make[2]: *** [Makefile:204: iotk_print_kinds.x] Error 1
$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-ld -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -o iotk_print_kinds.x iotk_print_kinds.o libiotk.a  
```

**Comment by wolfv** [comment]
@certik and me tried to help out on this PR by building it locally and seeing if we can find the issue. It looks like the linker is called with LDFLAGS that are supposed to go into the compiler (prefixed by `-Wl, ...`) and then the linker doesn't know how to handle those flags.

We're not sure what the correct fix is ... as the call to the linker seems hard coded inside the iotk package somehow? If we change the flags to not include `-Wl,` the configure fails..

Maybe iotk could be excluded from the build? 

Cheers!

**Comment by jan-janssen** [comment]
I am not an expert for quantum espresso, so feel free to modify the pull request. Talking to local quantum espresso users, they recommended to use the intel compiler instead.

**Comment by wolfv** [comment]
just as a FYI the following command seems to work (and link) fine:

```
gfortran -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -o iotk_print_kinds.x iotk_print_kinds.o libiotk.a
```

So basically if we can patch the makefile to use the gfortran compiler instead of the linker directly, it should work. 

However, the library is actually kind of crazy in that it ships with a tar archive that's unpacked from the Makefile ... so we need to figure out how to patch the unpacked files ... quite messed up if you ask me :)

**Comment by jan-janssen** [comment]
I guess we could download iotk from https://github.com/QEF/iotk/releases build a separate package, patch it, compile it, install it and then link to the library during the quantum espresso installation.

**Comment by wolfv** [comment]
I looked a bit more into it.

With the following argument in build.sh it does compile further than iotk:

make all LD="mpifort -fopenmpi"

(this will use the mpifort compiler as a linker and then the linking flags are valid). 

But I also needed this patch to correclty link to openmp or something:

```patch
diff --git a/PW/src/Makefile b/PW/src/Makefile
index 7ef9c1909..633ea03f1 100644
--- a/PW/src/Makefile
+++ b/PW/src/Makefile
@@ -254,7 +254,7 @@ all : tldeps pw.x generate_vdW_kernel_table.x generate_rVV10_kernel_table.x
 
 pw.x : $(PWOBJS) libpw.a $(LIBOBJS) $(QEMODS)
 	$(LD) $(LDFLAGS) -o $@ \
-	   $(PWOBJS) libpw.a $(QEMODS) $(LIBOBJS) $(QELIBS)
+	   $(PWOBJS) libpw.a $(QEMODS) $(LIBOBJS) -lfftw3_threads $(QELIBS) -lgomp
 	- ( cd ../../bin; ln -fs ../PW/src/$@ . ; \
                ln -fs ../PW/src/$@ dist.x ; ln -fs ../PW/src/$@ manypw.x ; )
 # dist.x and manypw.x link the same executable pw.x
```

This seems to work somewhat, but some other packages fails down the line with a similar error. So I am suspectign that more makefiles need to be patched like this. 

Another weird thing is that this lib seems to ship with it's own LAPACK. not sure if that's a good thing...
Anyways I hope this might help someone who knows more about quantum espresso to figure this out.

**Comment by certik** [comment]
Thanks @wolfv for looking into this. It seems the Quantum Espresso's build system is a mess, so the upstream package should clean this up.

File: recipes/qe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  license_file: LICENSE
+  summary: 'QUANTUM ESPRESSO (opEn-Source Package for Research in Electronic Structure, Simulation, and Optimization)'
+
+  description: |
+    QUANTUM ESPRESSO is an integrated suite of Open-Source computer 
+    codes for electronic-structure calculations and materials modeling 
+    at the nanoscale. It is based on density-functional theory, plane 
+    waves, and pseudopotentials.
+  doc_url: http://www.quantum-espresso.org/Doc/user_guide/
+  dev_url: https://github.com/simplejson/simplejson
```
Comment (hmaarrfk) [inline recipes/qe/meta.yaml]: this probably needs to be removed or updated

File: recipes/qe/License [hunk 2]
```diff
@@ -0,0 +1,340 @@
+		    GNU GENERAL PUBLIC LICENSE
```
Comment (hmaarrfk) [inline recipes/qe/License]: WHy is this not taken from the source?

**Comment by certik** [comment]
@jan-janssen do you plan to submit another PR for Quantum Espresso, or are we giving up?

**Comment by jan-janssen** [comment]
I am still interested - but currently I have to focus on other projects maybe in October. ... if somebody else wants to give it a go, feel free.

**Pull request closed without merging**